src/main/java/com/faforever/client/patch/GameBinariesUpdateTaskImpl.java

```diff
@@ -107,11 +110,16 @@ private void downloadFafExeIfNecessary(Path exePath) throws IOException {
             .listener(this::updateProgress)
             .copy();
       }
+      makeSureExeIsExecutable(exePath);
     } finally {
       ResourceLocks.freeDownloadLock();
     }
   }
 
+  private void makeSureExeIsExecutable(Path exePath) {
+    exePath.toFile().setExecutable(true);
```

Comment:
converting paths->file, file->path is a bad practice, nod need for that here, just use the util method from the Files class that takes a path as an argument:

`Files.setPosixFilePermissions(filePath, Sets.immutableEnumSet(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_EXECUTE));`

Reply:
also, please check system is unix before setting the executable bit